sound/soc/sof/core.c

```diff
@@ -177,42 +224,9 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 		goto ipc_err;
 	}
 
-	/* load the firmware */
-	ret = snd_sof_load_firmware(sdev);
-	if (ret < 0) {
-		dev_err(sdev->dev, "error: failed to load DSP firmware %d\n",
-			ret);
+	ret = sof_load_and_run_firmware(sdev);
```

Comment:
that sounds fine, but does it make sense to call snd_sof_dbg_init() for virtualized stuff?

It seems the only thing we should care about is the IPC init, no?